core/dbt/compilation.py

```diff
@@ -371,7 +371,7 @@ def _recursively_prepend_ctes(
 
             _extend_prepended_ctes(prepended_ctes, new_prepended_ctes)
 
-            new_cte_name = self.add_ephemeral_prefix(cte_model.name)
+            new_cte_name = self.add_ephemeral_prefix(cte_model.alias)
```

Comment:
It strikes me as a bit fragile that this line of code and the `BaseRelation.add_ephemeral_prefix()` implementation in `dbt-adapters` are both responsible for generating the CTE identifier for ephemeral models; ideally this bugfix could have been made in one PR against one of either `dbt-core` or `dbt-adapters` without requiring PRs against both repos. I wonder if perhaps there's a way to transfer responsibility over this call to the adapter? Happy to make that change if you can think of a reasonable approach.